typed-racket-lib/typed-racket/typecheck/tc-envops.rkt

```diff
@@ -56,6 +56,18 @@
         [else (let ([flds* (append lhs (cons (make-fld ty* acc-id #f) (cdr rhs)))])
                 (make-Struct nm par flds* proc poly pred))]))]
 
+    ;; class field ops
+    ;;
+    ;; This is a bit of an abuse of filters/objects. Normally, the object in
+    ;; an expression like (car? x) is the `x' identifier. However, for an
+    ;; application of a field accessor (accessor obj), the id `accessor` is
+    ;; the object. We need this so that we can refine the *range* of the
+    ;; accessor, not the object's type.
```

Comment:
This comment confuses me -- maybe I'm reading it wrong. 

'car?' should be 'car', right? and so we're saying 'car' is the new path element into the object 'x'?

Is `accessor' the object? or the path element added to the object of 'obj'?